modules/local/locidex_select.nf

```diff
@@ -0,0 +1,208 @@
+/*
+Locidex provides the option to select a database from a group of databases, by
+using a manifest file. To prevent copying all databases each time an allele database
+needs to be selected this file will read only the "manifest" file of the databases and
+to pick the correct allele scheme.
+
+
+Locidex Manifest is setup as:
+    {
+        db_name: [
+            {
+                "path": path/to/db/relative/to/manifest
+                # DB Config data, the newest db data will be selected as versions are not standardized
+                "config": {
+                    "db_name": "Locidex Database 1",
+                    "db_version": "1.0.0",
+                    "db_date": "yyyy-MM-dd",
+                    "db_author": "test1",
+                    "db_desc": "test1",
+                    "db_num_seqs": 53,
+                    "is_nucl": true,
+                    "is_prot": true,
+                    "nucleotide_db_name": "nucleotide",
+                    "protein_db_name": "protein"
+                }
+            }
+        ]
+    }
+*/
+
+import groovy.json.JsonSlurper
+import groovy.json.JsonBuilder
+import java.text.SimpleDateFormat
+
+
+process LOCIDEX_SELECT {
+    tag "$meta.id"
+    label "process_single"
+
+    input:
+    tuple val(meta), val(top_hit), val(contigs)
+    val manifest // This is a json file to be parsed
+
+    output:
+    tuple val(meta), val(contigs), val(scheme), val(paired_p), emit: db_data
+    tuple val(meta), path(output_config), emit: config_data
+
+    exec:
+    if(params.allele_scheme == null && params.locidex.allele_database == null){
+        exit 1, "Allele calling is enabled but there is no allele scheme or locidex allele database location present."
+    }
+
+    // Tokenize the "top_hit" or species value to identify all relevant match parts of the string
+    def species_data = top_hit.split('_|\s')
+    species_data = species_data*.toLowerCase()
+
+    // De-serialize the manifest file from the database location
+    def jsonSlurper = new JsonSlurper()
+    String json_data = manifest.text
+    def allele_db_data = jsonSlurper.parseText(json_data)
+    def allele_DB_KEYs = allele_db_data.keySet() as String[]
+
+    // Tokenize all database keys for lookup of species top hit in the database names
+    def databases = []
+    def shortest_entry = Integer.MAX_VALUE
+    for(allele_db in allele_DB_KEYs){
+        def db_tokens = allele_db.split('_|\s')
+        for(token in db_tokens){
+            def tok_size = token.size()
+            if(tok_size < shortest_entry){
+                shortest_entry = tok_size
+            }
+        }
+        databases.add(new Tuple(db_tokens*.toLowerCase(), allele_db))
+    }
+
+    def DB_TOKES_POS = 0
+    def DB_KEY = 1
+
+    // Remove spurious characters from tokenized string
+    species_data = species_data.findAll { it.size() >= shortest_entry }
```

Comment:
If you change the `databases` loop to use the collect function, `species_data` will also have to be updated to something like:
	`def shortestTokenSize = databases.collect { it[2] }.min()`
`species_data = species_data.findAll { it.size() >= shortestTokenSize }`

Reply:
For the above comment, I did not use exactly what was put there, but I did do something inspired by it!! I think it is alot cleaner now.